engine/src/test/java/org/hibernate/search/testsupport/TestConstants.java

```diff
@@ -89,4 +85,11 @@ public static String getIndexDirectory(Class<?> testClass) {
 	public static boolean arePerformanceTestsEnabled() {
 		return Boolean.getBoolean( "org.hibernate.search.enable_performance_tests" );
 	}
+
+	private static class CallerProvider extends SecurityManager {
+
+		public Class<?> getCallerClass() {
+			return getClassContext()[2];
```

Comment:
Let's assume Surefire's Fork is disabled: the same JVM would run all modules and initialize the static fields of this class only once.
The value of TestConstants.callerProvider would be initialized once, at first use, and that might be pointing to the target directory of a different module than the one being tested.